### PR TITLE
Allow CausalityBreakpoint

### PR DESCRIPTION
**Pull request by cston**
**Customer scenario**

Stepping into web service code fails with error dialog.

**Bugs this fixes:** 

366999

**Workarounds, if any**

No workaround.

**Risk**

Low

**Performance impact**

Low

**Is this a regression from a previous update?**

Yes, regression from previous update.

**How was the bug found?**

Found in product testing.

### CODE REVIEW

**Comment by cston** [comment]
@plnelson, @kevinh-ms, @dotnet/roslyn-interactive please review.

File: src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs [hunk 1]
```diff
@@ -117,11 +118,10 @@ internal void OnModuleLoad(TProcess process, TModule module, OnFunctionResolvedD
...
         {
             var languageId = GetLanguageId(request);
-            if (languageId == Guid.Empty)
-            {
-                return true;
-            }
-            return languageId == LanguageId;
+            return languageId == Guid.Empty ||
+                languageId == LanguageId ||
+                languageId == DkmLanguageId.CausalityBreakpoint ||
```
Comment (tmat) [inline src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs]: Perhaps add some comment on what Causality Breakpoint is and why we are handling it.

**Comment by tmat** [comment]
👍

File: src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs [hunk 2]
```diff
@@ -117,11 +118,10 @@ internal void OnModuleLoad(TProcess process, TModule module, OnFunctionResolvedD
...
+                languageId == DkmLanguageId.MethodId;
```
Comment (plnelson) [inline src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs]: The new BP binder should ignore DkmLanguageId.MethodId.  These are handled by the Causality Monitor.

File: src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs (hunk 1 above)
Comment (cston) [inline src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs]: Added comment.

File: src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs (hunk 2 above)
Comment (cston) [inline src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs]: Removed `DkmLanguageId.MethodId` check.

**Comment by plnelson** [comment]
👍

**Comment by cston** [comment]
@mattgertz for approval.

**Comment by KevinH-MS** [comment]
Late, but 👍

**Pull request merged**